src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        int index = random.nextInt(Figures.values().length);
+        Figures figure = Figures.values()[index];
+
+        switch (figure.toString()) {

```

Comment:
Instead of using `toString()`, use `name()` to get the string representation of the enum constant. This is a more reliable method as per the checklist guidelines.